common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java

```diff
@@ -47,13 +47,20 @@ public static int roundNumberOfBytesToNearestWord(int numBytes) {
   public static boolean arrayEquals(
       Object leftBase, long leftOffset, Object rightBase, long rightOffset, final long length) {
     int i = 0;
-    while (i <= length - 8) {
-      if (Platform.getLong(leftBase, leftOffset + i) !=
-        Platform.getLong(rightBase, rightOffset + i)) {
-        return false;
-      }
-      i += 8;
-    }
+
+      // This attempts to speed up the memcmp type of operation, but there is no way
+      // to guarantee that the offsets will be on a word boundary in order to use
+      // Platform.getLong
```

Comment:
I get it. Well at least, why not leave the current behavior for non-SPARC architectures?

Also, I dunno, is it that rare that the offsets don't match mod 8? should happen 1/8th of the time if the offsets are randomly distributed and I'm not even sure that's true. Avoiding a byte by byte comparison is probably a big win for any non-trivial region, and worth checking a simple condition.